### ISSUE TITLE
Obsidian-wordpress deletes Obsidian note frontmatter on publishing

### ISSUE THREAD

**Issue by OneRadRedDad** [devbean/obsidian-wordpress#71]
### What Happens

If I use the Wordpress publish button in Obsidian's sidebar to publish a note, my frontmatter notes are replaced with frontmatter generated by Obsidian-wordpress.

### What I Expect to Happen

I expect Obsidian-wordpress to append its generated frontmatter to the end of what I already have in the document.

### Workflow to reproduce this issue:

1. Open Obsidian.md
2. Navigate to any obsidian note with frontmatter in the publisher
```
Frontmatter example:
---
ImportedOn: Saturday, 18 December 2021 8:41:47 PM
Tags: Category/Mundane-Item
---
```
3. Click the Wordpress button in the Obisidan sidebar
4. Click **Publish**

The note then then loses its existing Frontmatter and gains the following:
```
postId: 20
postType: post
categories:
  - 1
---
```

**Comment by antonvh** [comment]
Yes. I have the same problem! I prefer to have YAML data merged instead of replaced, because I use the metadata for more than wordpress info. At least the option to not update the metadata should be nice.

Can you point me to the bit of code that reads and replace the frontmatter metadata? Maybe I can try to fix it in a PR. I would also like to add seo snippet and keyphrase through this plugin.

Great plugin, though!

**Comment by ianbetteridge** [comment]
Same problem here too, which is a pain as I use templates and front matter to automatically move content around folders in Obsidian.

**Comment by alexture** [comment]
Same issue here, it's my main pain point with this otherwise amazing extension. :)

**Comment by pawel-domanski** [comment]
I have the same issue :(

**Comment by Kjeld76** [comment]
Still happens in 0.18.0.

I hit this twice in a row last week, and I think it's worse than "replaced". Look at the block in the first post — it starts at `postId:`, there's no opening `---`. That's the part that got me. Without the opening delimiter Obsidian doesn't see frontmatter at all anymore, it just renders the whole thing as body text. Properties disappear from the sidebar, queries stop matching the note, and templates that read those fields break.

First publish lost `title` and `tags` and mangled the delimiters. Second publish on the same note removed the opening `---` and everything except `categories`. I ran `od -c` on the file afterwards because I didn't quite believe it, and yes, it really does start straight at `categories`:

```
categories:
  - 1
---
```

Plugin 0.18.0, REST API with an application password, WordPress 7.1, Windows.

What makes it easy to miss is that publishing itself works perfectly. The post shows up in WordPress exactly as expected, so nothing signals that the local note just lost most of its metadata. If you edit and republish the same note regularly, it adds up quietly.

Looks like the same root cause as #65 and #91.

For what it's worth, writing back only the keys the plugin owns instead of regenerating the whole block would avoid most of this — and refusing to write when the result wouldn't parse as YAML would turn a data loss into a harmless abort.
